asllib/StaticEnv.mli

```diff
@@ -25,15 +25,18 @@ open ASTUtils
 
 (** Static Environment used for type-checking (cf {!Typing}. *)
 
+type func = int ref * AST.func
```

Comment:
So this allows referring to each subprogram uniquely?
Is this instead of the name mangling for overloaded functions (that is, naming them foo1, foo2, etc.)?
Why do we need a reference though?